collaboration-service/index.js

```diff
@@ -36,11 +36,26 @@ io.on('connection', (socket) => {
     socket.on(event, () => {
       socket.rooms.forEach((room) => {
         if (room !== socket.id) {
-          // Emit to other sockets in the same room this socket had joined
-          socket
-            .to(room)
-            .emit('session-end', 'Your peer had left the session.', 'warning');
-          io.socketsLeave(room);
+          setTimeout(function () {
+            if (
+              event === sessionEndEvents[0] ||
+              // If after 5 seconds, the room is still less than 2 people,
+              // we can assume that the user had closed the tab.
+              // Otherwise, the user just refreshed the page
+              !io.sockets.adapter.rooms.get(room) ||
+              io.sockets.adapter.rooms.get(room).size < 2
+            ) {
+              // Emit to other sockets in the same room this socket had joined
+              socket
+                .to(room)
+                .emit(
+                  'session-end',
+                  'Your peer had left the session.',
```

Comment:
Just a grammar mistake: "Your peer has left the session."
Same below "You have left the session."

Reply:
Oh thanks man, sorry my English is pretty bad haha